File: Arduino/Version5/Python Code/src/FirmwareSettings.py

class FirmwareSettings:
    def __init__(self):
        """ Create default Settings Object
            DO NOT TOUCH
        """
        self.defaultSettings = {
            "FIRMWARE_VERSION": {"id": 0, "value": 5}, # this will be discarded if we try to change it. it will just return the on-device firmware version
            "logPerformanceMetrics": {"id": 1, "value": False},
            "enableSensReduction": {"id": 2, "value": True},
            "sensReductionDurationMilliseconds": {"id": 3, "value": 12000},
            "sensReductionAmmountX": {"id": 4, "value": 60}, # 100 = full movement (no reduction) 0 = full lockout (full reduction)
            "sensReductionAmmountY": {"id": 5, "value": 60}, # 100 = full movement (no reduction) 0 = full lockout (full reduction)
            "enableSpinning": {"id": 6, "value": False},
            "spinAmountPerRotation": {"id": 7, "value": 120},
            "spinNumberOfRotations": {"id": 8, "value": 8},
            "spinDelayBetweenRotationsMilliseconds": {"id": 9, "value": 9},
            "spinLockoutMouse": {"id": 10, "value": False},
            "spinBeforeAfterMouseEvent": {"id": 11, "value": 2}, # 0=Before, 1=After, 2=Both
            "disablePassthroughForMMB": {"id": 12, "value": 1}, # 0 = False, 1 = True, 2 = Only if MMB has been pressed within duration, 3 = Only pass through if double tapped (and or held) within duration
            "disablePassthroughForRMB": {"id": 13, "value": 3},
            "disablePassthroughForLMB": {"id": 14, "value": 0},
            "disablePassthroughForMB4": {"id": 15, "value": 2},
            "disablePassthroughForMB5": {"id": 16, "value": 2},
        }

        self.mySettings = {
            "FIRMWARE_VERSION": {"id": 0, "value": 5}, # this will be discarded if we try to change it. it will just return the on-device firmware version
            "logPerformanceMetrics": {"id": 1, "value": False},
            "mouseLockoutEnable": {"id": 2, "value": True},
            "mouseLockoutDurationMilliseconds": {"id": 3, "value": 8},
            "enableSpinning": {"id": 4, "value": 0}, # 0=Disabled, 1=Enabled
            "spinAmountPerRotation": {"id": 5, "value": 5357},
            "spinNumberOfRotations": {"id": 6, "value": 6},
            "spinDelayBetweenRotationsMilliseconds": {"id": 7, "value": 16},
            "spinLockoutMouseUntilCompletion": {"id": 8, "value": True},
            "spinBeforeAfterMouseEvent": {"id": 9, "value": 2}, # 0=Before, 1=After, 2=Both
            "disablePassthroughForMMB": {"id": 10, "value": 1}, # 0 = False, 1 = True, 2 = Only if MMB has been pressed within duration
            "disablePassthroughForRMB": {"id": 11, "value": 0},
            "disablePassthroughForLMB": {"id": 12, "value": 0},
            "disablePassthroughForMB4": {"id": 13, "value": 1},
            "disablePassthroughForMB5": {"id": 14, "value": 1},
        }

File: Arduino/streamcheats firmware/Arduino Code/Spoof - Synaptics Generic HID/Viper V3 Pro/MouseInTheMiddle/FirmwareSettings.cpp
#include "FirmwareSettings.h"
#include "Config.h" // Include if you need constants or global configurations

FirmwareSettings::FirmwareSettings() {}

void FirmwareSettings::updateSettings(const uint8_t* data) {
    uint8_t settingId = data[0];
    int16_t settingValue = data[1] | (data[2] << 8);  // Combine two bytes into a 16-bit integer

    switch (settingId) {
        case 0:
            FTDI_DEVICE.print("V: ");
            FTDI_DEVICE.println(FIRMWARE_VERSION);
            break;
        case 1:
            logPerformanceMetrics = settingValue;
            printSettingChange("logPerformanceMetrics", String(logPerformanceMetrics));
            break;
        case 2:
            enableSensReduction = settingValue;
            printSettingChange("enableSensReduction", String(enableSensReduction));
            break;
        case 3:
            sensReductionDurationMilliseconds = settingValue;
            printSettingChange("sensReductionDurationMilliseconds", String(sensReductionDurationMilliseconds));
            break;
        case 4:
            sensReductionAmmountX = settingValue;
            printSettingChange("sensReductionAmmountX", String(sensReductionAmmountX));
            break;
        case 5:
            sensReductionAmmountY = settingValue;
            printSettingChange("sensReductionAmmountY", String(sensReductionAmmountY));
            break;
        case 6:
            enableSpinning = settingValue;
            printSettingChange("enableSpinning", String(enableSpinning));
            break;
        case 7:
            spinAmountPerRotation = settingValue;
            printSettingChange("spinAmountPerRotation", String(spinAmountPerRotation));
            break;
        case 8:
            spinNumberOfRotations = settingValue;
            printSettingChange("spinNumberOfRotations", String(spinNumberOfRotations));
            break;
        case 9:
            spinDelayBetweenRotationsMilliseconds = settingValue;
            printSettingChange("spinDelayBetweenRotationsMilliseconds", String(spinDelayBetweenRotationsMilliseconds));
            break;
        case 10:
            spinLockoutMouseUntilCompletion = settingValue;
            printSettingChange("spinLockoutMouseUntilCompletion", String(spinLockoutMouseUntilCompletion));
            break;
        case 11:
            spinBeforeAfterMouseEvent = settingValue;
            printSettingChange("spinBeforeAfterMouseEvent", String(spinBeforeAfterMouseEvent));
            break;
        case 12:
            disablePassthroughForMMB = settingValue;
            printSettingChange("disablePassthroughForMMB", String(disablePassthroughForMMB));
            break;
        case 13:
            disablePassthroughForRMB = settingValue;
            printSettingChange("disablePassthroughForRMB", String(disablePassthroughForRMB));
            break;
        case 14:
            disablePassthroughForLMB = settingValue;
            printSettingChange("disablePassthroughForLMB", String(disablePassthroughForLMB));
            break;
        case 15:
            disablePassthroughForMB4 = settingValue;
            printSettingChange("disablePassthroughForMB4", String(disablePassthroughForMB4));
            break;
        case 16:
            disablePassthroughForMB5 = settingValue;
            printSettingChange("disablePassthroughForMB5", String(disablePassthroughForMB5));
            break;
        default:
            FTDI_DEVICE.print("Unknown setting ID received -> ");
            FTDI_DEVICE.println(settingId);
            FTDI_DEVICE.print("Unknown setting ID Value received -> ");
            FTDI_DEVICE.println(settingValue);
            break;
    }
}

void FirmwareSettings::printSettingChange(const String& settingName, const String& value) const {
    FTDI_DEVICE.print("Setting changed - ");
    FTDI_DEVICE.print(settingName);
    FTDI_DEVICE.print(": ");
    FTDI_DEVICE.println(value);
}
